onadata/apps/logger/models/xform.py

```diff
@@ -616,15 +626,33 @@ def _check_version_set(self, survey):
         the default version in this datetime (yyyymmddhhmm) format.
         """
 
+        change_version = False
         # get the json and check for the version key
         survey_json = json.loads(survey.to_json())
-        if not survey_json.get("version"):
+        version = survey_json.get("version")
+        if version is None:
             # set utc time as the default version
-            survey_json['version'] = \
-                datetime.utcnow().strftime("%Y%m%d%H%M")
+            change_version = True
+            version = create_xform_version()
+        else:
+            try:
+                int(version)
+            except ValueError:
+                raise XLSFormError(
+                    _(u"version should be a string of upto 10 numbers that "
+                      "describe this revision. You could use a string of the "
+                      "form 'yyyymmddrr'. For example, 2017021501 is the 1st "
+                      "revision from Feb 15th, 2017."))
+            else:
+                if len(version) > 10:
+                    change_version = True
+
+        if change_version:
+            survey_json['version'] = version[:10]
```

Comment:
What case is this handling? Would it perhaps to also throw an error here? I'm thinking of someone who mistakenly using a 16 digit version with hours, minutes, and seconds in their format. We'd silently truncate it and end up w/overlapping versions